### PR TITLE
Prevent localStorage errors by verifying its existence

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
## Description:
- **What's changed?** Verify that localStorage exists before querying it
### Resolves:

* [OKTA-480338](https://oktainc.atlassian.net/browse/OKTA-480338)

### CODE REVIEW

File: packages/@okta/vuepress-theme-prose/util/coveo/endpoint.js [hunk 1]
```diff
@@ -2,13 +2,15 @@ import axios from "axios";
...
+import storage from '../localStorage'
+
 const TOKEN_ENDPOINT =
   "https://wt1ugse0be.execute-api.us-west-2.amazonaws.com/prod/token?site=developer";
 const COVEO_PIPELINE = "oktaproduction9ounvcxa";
 const COVEO_ENDPOINT = "https://platform.cloud.coveo.com/rest/search";
 
 const _getToken = () => {
-  let token = localStorage.getItem("coveo_token");
+  let token = storage.getItem("coveo_token");
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/util/coveo/endpoint.js:13]: maybe also move `coveo_token` to a constant

File: packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue [hunk 2]
```diff
@@ -24,20 +25,21 @@ export default {
 
   methods: {
     getDarkMode: function() {
-      this.isDarkMode = localStorage[themeModeCookieName] 
-          ? JSON.parse(localStorage[themeModeCookieName])  
+      this.isDarkMode = storage.getItem(THEME_MODE_KEY) 
+          ? JSON.parse(storage.getItem(THEME_MODE_KEY))  
           : window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue:30]: while we're watching for compatible code, `matchMedia` is fairly well supported but not available on IE9: https://caniuse.com/mdn-api_window_matchmedia

Not sure if we care, but we could add safety around this as well

File: packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue [hunk 3]
```diff
@@ -24,20 +25,21 @@ export default {
...
+      this.isDarkMode = !this.isDarkMode;
+      storage.setItem(THEME_MODE_KEY, this.isDarkMode);
       this.addHtmlClass();
     },
 
     addHtmlClass: function() {
       const bodyClasses = document.body.classList;
       bodyClasses.add('loaded');
-      (this.isDarkMode === true) ? bodyClasses.add(darkThemeHtmlClass) : bodyClasses.remove(darkThemeHtmlClass);
+      (this.isDarkMode === true) ? bodyClasses.add(DARK_THEME_CLASS) : bodyClasses.remove(DARK_THEME_CLASS);
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue:42]:
```suggestion
      this.isDarkMode ? bodyClasses.add(DARK_THEME_CLASS) : bodyClasses.remove(DARK_THEME_CLASS);
```

File: packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue [hunk 4]
```diff
@@ -24,20 +25,21 @@ export default {
 
   methods: {
     getDarkMode: function() {
-      this.isDarkMode = localStorage[themeModeCookieName] 
-          ? JSON.parse(localStorage[themeModeCookieName])  
+      this.isDarkMode = storage.getItem(THEME_MODE_KEY) 
+          ? JSON.parse(storage.getItem(THEME_MODE_KEY))
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-theme-prose/components/ThemeModeToggle.vue:29]: `JSON.parse` could throw if somehow this storage item is polluted or corrupted

**Review by shawyu-okta — APPROVED** [review]
just a few comments but overall lgtm

**Comment by francoislehoux-okta** [comment]
I'll fix the two quick ones, I need to move onto the RU and BY ban asap. We can open a Jira for the other two

**Pull request merged**